tests/test_api_helpers.py

```diff
@@ -56,44 +56,59 @@ def test_combo_simple():
     assert [Modifier.ALT] == combo.modifiers
 
 def test_type_simple():
-    out = type("hello5")
+    out = to_keystrokes("hello5")
     assert [Key.H, Key.E, Key.L, Key.L, Key.O, Key.KEY_5] == out()
 
 def test_type_simple_with_shift():
-    out = type("Hello")
+    out = to_keystrokes("Hello")
     assert [C("Shift-H"), Key.E, Key.L, Key.L, Key.O] == out()
 
 def test_type_unsupported_character():
     with pytest.raises(CharacterNotSupported) as e:
-        out = type("{")
+        out = to_keystrokes("{")
         assert e.message == "The character { is not supported by `type` yet"
 
 def test_type_too_long():
     with pytest.raises(TypingTooLong) as e:
-        out = type("lasdjlkad jlkasjd laksjdlkasj dlkasj dlk ajlkd jaldkjal"
+        out = to_keystrokes("lasdjlkad jlkasjd laksjdlkasj dlkasj dlk ajlkd jaldkjal"
             "asdkjhasdkjahkjdhaskjdhakjdhkjadh kajhdkjashdkashdkjajhdajksd")
         assert e.message == "`type` only supports strings of 100 characters or less"
 
 def test_type_extended_ascii():
-    out = type("\u00ff")
+    out = to_keystrokes("\u00ff")
     assert [ C("Shift-Ctrl-U"),
              Key.F, Key.F,
              Key.ENTER] == out()
 
 def test_ascii_keys():
-    out = type("`-=[]\\;',./")
+    out = to_keystrokes("`-=[]\\;',./")
     assert [ Key.GRAVE, Key.MINUS, Key.EQUAL, Key.LEFT_BRACE,
              Key.RIGHT_BRACE, Key.BACKSLASH, Key.SEMICOLON,
              Key.APOSTROPHE, Key.COMMA, Key.DOT, Key.SLASH
              ] == out()
 
 def test_type_unicode():
-    out = type("🎉")
+    out = to_keystrokes("🎉")
     assert [ C("Shift-Ctrl-U"),
              Key.KEY_1, Key.F, Key.KEY_3, Key.KEY_8, Key.KEY_9,
              Key.ENTER] == out()
 
-    out = type("\U0001f389")
+    out = to_keystrokes("\U0001f389")
     assert [ C("Shift-Ctrl-U"),
              Key.KEY_1, Key.F, Key.KEY_3, Key.KEY_8, Key.KEY_9,
              Key.ENTER] == out()
+
+def test_uncode_keystrokes():
+    out = unicode_keystrokes(0x00ff)
+    assert [ C("Shift-Ctrl-U"),
+             Key.F, Key.F,
+             Key.ENTER] == out()
+
+    out = unicode_keystrokes(0x10fad)
+    assert [ C("Shift-Ctrl-U"),
+             Key.KEY_1, Key.KEY_0, Key.F, Key.A, Key.D,
+             Key.ENTER] == out()
+
+    with pytest.raises(UnicodeNumberToolarge) as e:
+        out = unicode_keystrokes(0x110000)
+        assert e.message == "too large for Unicode keyboard entry."
```

Comment:
This could probably be improved.

Reply:
Not sure if this is related, or even an actual bug or something I did wrong in the config: 

```
Traceback (while executing your config):
  File "/home/kris/.config/kinto/keyszer.py", line 405, in <module>
  File "/home/kris/Downloads/josh-keyszer/keyszer/src/keyszer/config_api.py", line 181, in unicode_keystrokes
if (n > 0x10ffff):
TypeError: '>' not supported between instances of 'str' and 'int'
```

Reply:
Of course I am. What kind of not-fool do you take me for? Remembered the function name but didn't change the strings. Fixed it.